src/msw/graphicsd2d.cpp

```diff
@@ -2474,9 +2478,9 @@ class wxD2DPenData : public wxGraphicsObjectRefData, public wxD2DManagedGraphics
     }
 
 private:
-    // We store the source pen for later when we need to recreate the
+    // We store the source info for later when we need to recreate the
     // device-dependent resources.
-    const wxPen m_sourcePen;
+    const wxGraphicsPenInfo& m_sourceInfo;
```

Comment:
This should be an object too, you seem to be using a dangling reference, i.e. this will result in an all but guaranteed runtime crash!